include/io.tcc

```diff
@@ -186,3 +190,128 @@ void IO<Tdim>::write_volumes(const std::map<unsigned, double>& volumes) {
   }
   std::cout << "Wrote Volumes \n";
 }
+
+//! \brief Output .vtk files for viewing material points
+//! \details Write material points coordinates and initial stresses
+//! \tparam Tdim dimension
+template <unsigned Tdim>
+void IO<Tdim>::write_vtk_stresses(
+    const std::vector<Eigen::VectorXd>& coordinates,
+    const std::vector<Eigen::VectorXd>& stresses) {
+
+  //! Check number of points match
+  unsigned num_coordinates, num_stresses;
+  unsigned num_points = 0;
+
+  num_coordinates = coordinates.size();
+  num_stresses = stresses.size();
+
+  if (num_coordinates == num_stresses) {
+    num_points = num_coordinates;
+  } else {
+    std::cerr << "Vector size of coordinates and stresses do not match\n";
```

Comment:
So? what do you want to do?

Reply:
actually I want it to exit... hmm, maybe I should use `catch` and `throw`?

Reply:
That won't be true, if you don't compute stresses.

Reply:
Unless you handle compute stresses or not differently.

Reply:
Why can't we just pass vector of elements here?

Reply:
We cannot pass vector of elements because you would need to generate your material points again, which might be different from what we generate using `material_points`

I am just thinking what will happen if I don't compute stress...

Reply:
What if write files in mesh? Would that be more elegant?

Reply:
Sorry, I should have said pass `material_points_` instead of `elements_`

Reply:
I thought we don't want to pass `material_points`? We don't want the pointer to be accessible right?

We can write .vtk in `mesh` as well. But I felt writing it in `io` made sense because it's an output. Elegant-wise, it might not, as we need to pass the directory and filename as well. What do you think?

Reply:
If we write it in the mesh, then we won't have to pass / create massive vectors, which is OK for small meshes, may not work for larger problems.

Reply:
Maybe it is better to call another write function in the mesh from the IO class with the filestream. This way we can avoid creating new vectors.

Reply:
Understood. Do you want the same for `coordinates` and `stresses` as well?

Reply:
Maybe that's better, would be a separate issue.

Reply:
So in `io`, I will call a function in `mesh` to write out part of the `vtk` right?

Reply:
Would I need to transfer `mesh` into `io`? That would defeat the purpose of doing it right?

Reply:
Ini `mpm-3d` the `ParticleCloud` owns `particles_` and thus this is no brainer to do it. But here, I would think that passing a vector would be the "cleanest". Passing `mesh` to `io` does not make sense. I could not think of a structure that would work (well, it's late here, maybe tomorrow morning I could haha)

Reply:
Refactored!